tenant/tenant.go

```diff
@@ -175,20 +173,13 @@ func (tsvc *TenantSvc) SetConfig(config common.ServiceConfig) error {
 	log.Println(config)
 	tsvc.config = config
 	storeConfig := config.ServiceSpecific["store"].(map[string]interface{})
-	storeType := strings.ToLower(storeConfig["type"].(string))
-	switch storeType {
-	case "mysql":
-		tsvc.store = &mysqlStore{}
-	case "mock":
-		tsvc.store = &mockStore{}
-	default:
-		return errors.New("Unknown store type: " + storeType)
-	}
-	return tsvc.store.setConfig(storeConfig)
+	tsvc.store = tenantStore{}
+	tsvc.store.ServiceStore = tsvc.store
```

Comment:
Good qestion, need a documentation for this pattern or antipattern. Added todo for now.